src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp

```diff
@@ -66,27 +522,397 @@ class boss_thorim : public CreatureScript
                 if (!UpdateVictim())
                     return;
-                //SPELLS @todo
 
-                //
+                events.Update(diff);
+
+                if (me->HasUnitState(UNIT_STATE_CASTING))
+                    return;
+
+                while (uint32 eventId = events.ExecuteEvent())
+                {
+                    switch (eventId)
+                    {
+                        case EVENT_SAY_AGGRO_2:
+                            Talk(SAY_AGGRO_2);
+                            break;
+                        case EVENT_SAY_SIF_START:
+                            if (Creature* sif = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_SIF)))
+                                sif->AI()->Talk(SAY_SIF_START);
+                            break;
+                        case EVENT_START_SIF_CHANNEL:
+                            if (Creature* sif = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_SIF)))
+                                sif->CastSpell(me, SPELL_TOUCH_OF_DOMINION);
+                            break;
+                        case EVENT_STORMHAMMER:
+                            DoCast(SPELL_STORMHAMMER);
+                            events.ScheduleEvent(EVENT_STORMHAMMER, urand(15000, 20000), 0, PHASE_1);
+                            break;
+                        case EVENT_CHARGE_ORB:
+                            DoCastAOE(SPELL_CHARGE_ORB);
+                            events.ScheduleEvent(EVENT_CHARGE_ORB, urand(15000, 20000), 0, PHASE_1);
+                            break;
+                        case EVENT_SUMMON_ADDS:
+                            SummonWave();
+                            events.ScheduleEvent(EVENT_SUMMON_ADDS, _orbSummoned ? 3000 : 10000, 0, PHASE_1);
+                            break;
+                        case EVENT_JUMPDOWN:
+                            if (_hardMode)
+                                if (Creature* sif = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_SIF)))
+                                    sif->AI()->DoAction(ACTION_START_HARD_MODE);
+                            me->RemoveAurasDueToSpell(SPELL_SHEATH_OF_LIGHTNING);
+                            me->SetReactState(REACT_AGGRESSIVE);
+                            SetCombatMovement(true);
+                            me->GetMotionMaster()->MoveJump(2134.79f, -263.03f, 419.84f, me->GetOrientation(), 30.0f, 20.0f);
+                            events.ScheduleEvent(EVENT_START_PERIODIC_CHARGE, 2000, 0, PHASE_2);
+                            events.ScheduleEvent(EVENT_UNBALANCING_STRIKE, 15000, 0, PHASE_2);
+                            events.ScheduleEvent(EVENT_CHAIN_LIGHTNING, 20000, 0, PHASE_2);
+                            break;
+                        case EVENT_UNBALANCING_STRIKE:
+                            DoCastVictim(SPELL_UNBALANCING_STRIKE);
+                            events.ScheduleEvent(EVENT_UNBALANCING_STRIKE, urand(15000, 20000), 0, PHASE_2);
```

Comment:
Since https://github.com/TrinityCore/TrinityCore/commit/2174c5d7dd9669384b2e68b8b406a3173042d7d9 
You can simplify to events.ScheduleEvent(EVENT_UNBALANCING_STRIKE, Seconds(8), Seconds(10)), for "random times"